From 6e451a6ac68550d566ce301cd5c3857115b9d7f1 Mon Sep 17 00:00:00 2001
From: "Arun P. Mohanan" <arun.p.m@linux.intel.com>
Date: Wed, 27 Jan 2021 18:22:58 +0530
Subject: [PATCH] Add state sensor messages to the registry

Add messages to registry to indicate state sensor state change.

Tested:
Build and redfish validator passes.
Logged these events and confirmed that they appear as expected on
Redfish.
GET: https://<BMC IP>/redfish/v1/Systems/system/LogServices/EventLog/Entries/1612528180
{
    "@odata.id": "/redfish/v1/Systems/system/LogServices/EventLog/Entries/1612528180",
    "@odata.type": "#LogEntry.v1_4_0.LogEntry",
    "Created": "2021-02-05T12:29:40+00:00",
    "EntryType": "Event",
    "Id": "1612528180",
    "Message": "Operational Fault Status of Card_health_1 state sensor changed from Error to Normal.",
    "MessageArgs": [
        "Operational Fault Status",
        "Card_health_1",
        "Error",
        "Normal"
    ],
    "MessageId": "OpenBMC.0.1.StateSensorNormal",
    "Name": "System Event Log Entry",
    "Severity": "OK"
}

Signed-off-by: Arun P. Mohanan <arun.p.m@linux.intel.com>
Signed-off-by: P Dheeraj Srujan Kumar <p.dheeraj.srujan.kumar@intel.com>
---
 .../registries/openbmc_message_registry.hpp   | 31 ++++++++++++++++++-
 1 file changed, 30 insertions(+), 1 deletion(-)

diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index 80d30c3..842115d 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -2173,6 +2173,36 @@ constexpr std::array registry = {
                      {},
                      "None.",
                  }},
+    MessageEntry{
+        "StateSensorNormal",
+        {
+            "Indicates that a state sensor has changed state to normal.",
+            "%1 of %2 state sensor changed from %3 to %4.",
+            "OK",
+            4,
+            {"string", "string", "string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "StateSensorWarning",
+        {
+            "Indicates that a state sensor has changed state to warning.",
+            "%1 of %2 state sensor changed from %3 to %4.",
+            "Warning",
+            4,
+            {"string", "string", "string", "string"},
+            "Check sensor subsystem for errors.",
+        }},
+    MessageEntry{
+        "StateSensorCritical",
+        {
+            "Indicates that a state sensor has changed state to critical.",
+            "%1 of %2 state sensor changed from %3 to %4.",
+            "Critical",
+            4,
+            {"string", "string", "string", "string"},
+            "Check sensor subsystem for errors.",
+        }},
     MessageEntry{"SystemInterfaceDisabledProvisioned",
                  {
                      "Indicates that the system interface is in the disabled "
@@ -2257,6 +2287,5 @@ constexpr std::array registry = {
             {"string"},
             "None.",
         }},
-
 };
 } // namespace redfish::registries::openbmc
-- 
2.17.1

